types/tx_msg.go

```diff
@@ -84,3 +84,6 @@ type TxDecoder func(txBytes []byte) (Tx, error)
 
 // TxEncoder marshals transaction to bytes
 type TxEncoder func(tx Tx) ([]byte, error)
+
+// Unmarshaler is a generic type for Unmarshal function
+type Unmarshaler func(bytes []byte, ptr interface{}) error
```

Comment:
Is this used somewhere else in the SDK? Maybe putting it only in `legacytx` makes more sense?

Reply:
It's used only in legacytx. It is a general type.

I've checked, there is no more similar types matching the following pattern:
```
"func\(.*\[\]byte,.*interface"
```
If you want I can move it to legacytx

Reply:
I think putting in `tx_msg.go` is definitely confusing: this function signature has nothing to do with Tx.

`legacytx` is the obvious place to put for me, but happy to put it somewhere else that makes more sense than `tx_msg.go`.

Reply:
OK, will move to legacytx, it will be easy to move it / create it if needed.